src/util/string.h

```diff
@@ -2,10 +2,18 @@
 
 #include <QCollator>
 #include <QColor>
+#include <QFileInfo>
 #include <QLocale>
+#include <QRegularExpression>
 #include <QString>
 #include <QStringRef>
 
+namespace {
+
+static const QRegularExpression kExtRgxp(R"(\(\*\.(.*)\)$)");
```

Comment:
I'm unsure whether the static makes a difference in this context. AFAIK globals are already static, especially if they're const. We don't use the static keyword for other globals anywhere else. My mistake should've clarified this in the previous comment. wdyt @uklotzde?
I also think a global deserves a better name. `kExtracExtensionRegex` comes to my mind.

Reply:
Good catch! I didn't notice this.

Moreover using an anonymous namespace in a header file is useless and wrong.

The const should get a better name and be declared in the `mixxx` namespace if it needs to be public. It must be declared as `extern` and defined separately in the .cpp file to create only a single instance per application.

Reply:
sure I can change it to `kExtracExtensionRegex`

I'll try to make sense of your comments -- keep in mind that after all I'm still a c++ entrepreneur ;)

Reply:
Since we agreed to put the function into a cpp file, some parts of uwes comments are irrelevant then. If you put the function and the regex into a cpp file, afaik the anonymous namespace should be enough. It also doesn't need to be public since its only used in that single function.

Reply:
Thanks for clarifying!